### PR TITLE
Add very basic docker support

### LINKED ISSUE #222: Add docker support
I think that adding docker support will be useful for a lot of users.

While shipping images on dockerhub with releases is ideal, we can still have a basic integration with mention in the docs & a dockerfile in the repo.

### PR DESCRIPTION
**Pull request by peauc**
Fix #222

Add basic docker support for self hosted build

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM golang:1.22.5-alpine
+
+WORKDIR /devzat
+
+COPY . .
+
+RUN go build
+
+EXPOSE 8080
```
Comment (Aeron) [inline Dockerfile:9]: The default port for the server is `2221`, and the alternative one is `8080`. Is there any specific reason why only the latter is specified? The `EXPOSE` instruction is only informative but still.

Comment (peauc) [reply Dockerfile:9]: During my tests i've found that I can connect to the chatroom using port 8080 so I replicated that in the dockerfile. I've spent no more than 15 minutes on the dockerfile + PR, no idea if other configurations works but interested in figuring it out

Comment (quackduck) [reply Dockerfile:9]: there's a bunch of ports used. you'd probably want to expose the rpc one and also both the ssh ones. please have a look at the source

**Comment by Arkaeriit** [comment]
Devzat uses the file system to store ssh keys, server configuration, and user configuration. All those should be permanent after a Devzat restart. I am not a Docker expert but it looks like the files are erased after closing the docker container.

**Comment by quackduck** [comment]
What's the relationship here with the other docker pr?

**Comment by peauc** [comment]
@quackduck there is none, im closing this one

**Pull request closed without merging**